tap_stripe/__init__.py

```diff
@@ -610,6 +612,12 @@ def sync_stream(stream_name, is_sub_stream=False):
         # when they are available via the API, so these streams will need
         # a short lookback window.
         if stream_name in IMMUTABLE_STREAMS:
+            if stream_name == "events":
+                # Start sync from 30 days before if bookmark/start_date is older than 30 days.
+                start_window = int(max(bookmark, (singer.utils.now() - timedelta(days=30)).timestamp()))
+                if start_window != bookmark:
+                    LOGGER.warning("Provided current bookmark/start_date is older than the last 30 days. So, starting sync for the last 30 days as Stripe Event API returns data for the last 30 days only.")
```

Comment:
Add unit test to verify this Warning message.

Reply:
Added unit test to verify Warning message.